packaging/makeself/install-or-update.sh

```diff
@@ -62,6 +77,14 @@ if [ ! "${DISABLE_TELEMETRY:-0}" -eq 0 ] ||
   REINSTALL_OPTIONS="${REINSTALL_OPTIONS} --disable-telemetry"
 fi
 
+if [ -n "${NETDATA_CERT_MODE}" ]; then
+  REINSTALL_OPTIONS="${REINSTALL_OPTIONS} --certificates ${NETDATA_CERT_MODE}"
+fi
+
+if [ -n "${NETDATA_CERT_TEST_URL}" ]; then
+  REINSTALL_OPTIONS="${REINSTALL_OPTIONS} --certificate-test-url ${NETDATA_CERT_TEST_URL}"
+fi
```

Comment:
This will fail

https://github.com/netdata/netdata/blob/9a7bfb9839e49137e63e181bbd86475ed64570d5/packaging/makeself/install-or-update.sh#L231

because `NETDATA_CERT_TEST_URL` contains slashes.

We need to use a different delimiter in the `sed` command.

Reply:
This works

```bash
sed -i "s|REINSTALL_OPTIONS=\".*\"|REINSTALL_OPTIONS=\"${REINSTALL_OPTIONS}\"|" "${NETDATA_PREFIX}/etc/netdata/.environment"
```

Reply:
Doesn't work. Let me help you and share chatGPT response:

---

The problem with using awk to escape slashes in your variable is that the backslashes themselves need to be escaped in the sed replacement string. This can become quite complex and tricky to manage.

An easier and more straightforward approach is to use a different delimiter in your sed command. Here's how you can do it without relying on awk:

```bash
REINSTALL_OPTIONS="--certificates check --certificate-test-url https://app.netdata.cloud"
sed "s|REINSTALL_OPTIONS=\".*\"|REINSTALL_OPTIONS=\"${REINSTALL_OPTIONS}\"|"

```

This script will work correctly as long as ${REINSTALL_OPTIONS} does not contain the chosen delimiter (| in this case).

If you need to handle escaping in a more complex scenario, you could use sed to escape the slashes. Here is how you can achieve that:

```bash
REINSTALL_OPTIONS="--certificates check --certificate-test-url https://app.netdata.cloud"
REINSTALL_OPTIONS_ESCAPED=$(printf '%s\n' "$REINSTALL_OPTIONS" | sed 's/[\/&]/\\&/g')
echo "${REINSTALL_OPTIONS_ESCAPED}"
sed "s/REINSTALL_OPTIONS=\".*\"/REINSTALL_OPTIONS=\"${REINSTALL_OPTIONS_ESCAPED}\"/"
```

---

So, if you insist on escaping:

```bash
REINSTALL_OPTIONS=$(printf '%s\n' "$REINSTALL_OPTIONS" | sed 's/[\/&]/\\&/g')
````

Reply:
So your solution is to go from an opaque AWK command to an arguably even _more_ opaque sed command?

Despite what ChatGPT says, this has nothing to do with the use of AWK, it’s an inherent issue in using _any_ external language in shell script, you need to escape things to one more level than you would normally need if working in that language directly, but you also need to remember what level of escaping you would need normally (which I forgot in this case). The fact that the escaping is done completely differently in sed does not solve that in any way, and the exact same mistake can just as easily be made there as well.

Reply:
I provided two solutions, both of which fix the issue I was testing for you.